### ISSUE TITLE
Allow custom Text Input

### ISSUE THREAD

**Issue by nandorojo** [CaioQuirinoMedeiros/react-native-currency-input#26]
I want to use my own input component. Ideally would be passed as a prop or a render function.

**Comment by nandorojo** [comment]
@CaioQuirinoMedeiros would you accept a PR if I sent one?

**Comment by nandorojo** [comment]
Here's my `patch` file for those interested:

```patch
diff --git a/node_modules/react-native-currency-input/lib/commonjs/CurrencyInput.js b/node_modules/react-native-currency-input/lib/commonjs/CurrencyInput.js
index 4aacbf6..2c004db 100644
--- a/node_modules/react-native-currency-input/lib/commonjs/CurrencyInput.js
+++ b/node_modules/react-native-currency-input/lib/commonjs/CurrencyInput.js
@@ -31,6 +31,7 @@ var _default = /*#__PURE__*/React.forwardRef((props, ref) => {
     minValue,
     signPosition = 'afterPrefix',
     showPositiveSign,
+    renderInput,
     ...rest
   } = props;
   const [startingWithSign, setStartingWithSign] = React.useState();
@@ -129,7 +130,7 @@ var _default = /*#__PURE__*/React.forwardRef((props, ref) => {
       signPosition
     }) : formattedValue;
   }, [formattedValue, prefix, signPosition, startingWithSign, suffix]);
-  return /*#__PURE__*/React.createElement(_reactNative.TextInput, _extends({
+  const nextProps = _extends({
     keyboardType: "numeric",
     selection: suffix ? {
       start: Math.max(textInputValue.length - suffix.length, 0)
@@ -138,7 +139,11 @@ var _default = /*#__PURE__*/React.forwardRef((props, ref) => {
     value: textInputValue,
     onChangeText: handleChangeText,
     ref: ref
-  }));
+  })
+  if (renderInput) {
+    return renderInput(nextProps);
+  }
+  return /*#__PURE__*/React.createElement(_reactNative.TextInput, nextProps);
 });
 
 exports.default = _default;
diff --git a/node_modules/react-native-currency-input/lib/module/CurrencyInput.js b/node_modules/react-native-currency-input/lib/module/CurrencyInput.js
index 6d1f5cd..c4c266a 100644
--- a/node_modules/react-native-currency-input/lib/module/CurrencyInput.js
+++ b/node_modules/react-native-currency-input/lib/module/CurrencyInput.js
@@ -17,6 +17,7 @@ export default /*#__PURE__*/React.forwardRef((props, ref) => {
     minValue,
     signPosition = 'afterPrefix',
     showPositiveSign,
+    renderInput,
     ...rest
   } = props;
   const [startingWithSign, setStartingWithSign] = React.useState();
@@ -115,7 +116,7 @@ export default /*#__PURE__*/React.forwardRef((props, ref) => {
       signPosition
     }) : formattedValue;
   }, [formattedValue, prefix, signPosition, startingWithSign, suffix]);
-  return /*#__PURE__*/React.createElement(TextInput, _extends({
+  const nextProps = _extends({
     keyboardType: "numeric",
     selection: suffix ? {
       start: Math.max(textInputValue.length - suffix.length, 0)
@@ -124,6 +125,10 @@ export default /*#__PURE__*/React.forwardRef((props, ref) => {
     value: textInputValue,
     onChangeText: handleChangeText,
     ref: ref
-  }));
+  })
+  if (renderInput) {
+    return renderInput(nextProps)
+  }
+  return /*#__PURE__*/React.createElement(TextInput, nextProps);
 });
 //# sourceMappingURL=CurrencyInput.js.map
\ No newline at end of file
diff --git a/node_modules/react-native-currency-input/lib/typescript/src/props.d.ts b/node_modules/react-native-currency-input/lib/typescript/src/props.d.ts
index 07c5c03..0013ba8 100644
--- a/node_modules/react-native-currency-input/lib/typescript/src/props.d.ts
+++ b/node_modules/react-native-currency-input/lib/typescript/src/props.d.ts
@@ -89,6 +89,8 @@ export interface CurrencyInputProps extends Omit<TextInputProps, 'value'> {
      * Use `showPositiveSign` if you want to show the `+` sign.
      */
     signPosition?: 'beforePrefix' | 'afterPrefix';
+
+    renderInput?: (props: TextInputProps) => JSX.Element;
 }
 export interface FakeCurrencyInputProps extends CurrencyInputProps {
     /**
diff --git a/node_modules/react-native-currency-input/src/CurrencyInput.tsx b/node_modules/react-native-currency-input/src/CurrencyInput.tsx
index 7626856..0ec89e9 100644
--- a/node_modules/react-native-currency-input/src/CurrencyInput.tsx
+++ b/node_modules/react-native-currency-input/src/CurrencyInput.tsx
@@ -18,6 +18,7 @@ export default React.forwardRef<TextInput, CurrencyInputProps>((props, ref) => {
     minValue,
     signPosition = 'afterPrefix',
     showPositiveSign,
+    renderInput,
     ...rest
   } = props;
 
@@ -157,18 +158,24 @@ export default React.forwardRef<TextInput, CurrencyInputProps>((props, ref) => {
       : formattedValue;
   }, [formattedValue, prefix, signPosition, startingWithSign, suffix]);
 
+  const nextProps: any = {
+    keyboardType: 'numeric',
+    selection: suffix ? 
+    { start: Math.max(textInputValue.length - suffix.length, 0) }
+    : props?.selection,
+    ...rest,
+    value: textInputValue,
+    onChangeText: handleChangeText,
+    ref
+  }
+
+  if (renderInput) {
+    return renderInput(nextProps);
+  }
+
   return (
     <TextInput
-      keyboardType="numeric"
-      selection={
-        suffix
-          ? { start: Math.max(textInputValue.length - suffix.length, 0) }
-          : props?.selection
-      }
-      {...rest}
-      value={textInputValue}
-      onChangeText={handleChangeText}
-      ref={ref}
+      {...nextProps}
     />
   );
 });
diff --git a/node_modules/react-native-currency-input/src/props.ts b/node_modules/react-native-currency-input/src/props.ts
index d9b56ee..473b3e4 100644
--- a/node_modules/react-native-currency-input/src/props.ts
+++ b/node_modules/react-native-currency-input/src/props.ts
@@ -108,6 +108,8 @@ export interface CurrencyInputProps extends Omit<TextInputProps, 'value'> {
    * Use `showPositiveSign` if you want to show the `+` sign.
    */
   signPosition?: 'beforePrefix' | 'afterPrefix';
+
+  renderInput?: (props: TextInputProps) => JSX.Element;
 }
 
 export interface FakeCurrencyInputProps extends CurrencyInputProps {
```

**Comment by CaioQuirinoMedeiros** [comment]
> @CaioQuirinoMedeiros would you accept a PR if I sent one?

Yes

**Comment by CaioQuirinoMedeiros** [comment]
Released v1.1.0 with `renderTextInput` prop
